### ISSUE TITLE
Small bug on rST parsing on pyplot API in Sphinx doc output

### ISSUE THREAD

**Issue by Naereen** [matplotlib/matplotlib#8074]
Hi,
The current version of the Sphinx HTML documentation of [`pyplot` API](matplotlib.org/api/pyplot_api.html) appear to contain some errors on the HTML produced from rST in the docstrings.

### Examples
Here are two screenshots of the buggy output, for `semilogx`,  `semilogy` and `spy`:
![bug matplotlib doc 1](https://cloud.githubusercontent.com/assets/11994719/22942940/dc59963a-f2eb-11e6-864e-7e9caa053d3d.png)
![bug matplotlib doc 2](https://cloud.githubusercontent.com/assets/11994719/22942939/dc58898e-f2eb-11e6-97ec-f86074828919.png)
![bug matplotlib doc 3](https://cloud.githubusercontent.com/assets/11994719/22943058/5b3178a6-f2ec-11e6-9621-52507f48a330.png)

This bug is *probably* present on (possibly many) others pages.

### Possible reason
My suspicion is that
```rst
:class:`~matplotlib.lines.Line2D`
```
is not interpreted correctly, and should probably simple be with `~`:
```rst
:class:`matplotlib.lines.Line2D`
```

**Comment by Naereen** [comment]
Note: before starting the locate the bug in the documentation or the docstring, understand the reason of the bug and try to fix it, I would prefer to have some feedback:
- is it a bug from Sphinx, or from the rST code ?
- is it already fixed in some waiting-to-be-merged PR ?

**Comment by tacaswell** [comment]
It is almost certainly a bug in the rst code and I have not seen a PR related to this go by.  However, I do not think it is the `~` as that is how to tell sphinx to include only the last part of the fully qualified name.

**Comment by Naereen** [comment]
OK. I was unsure of the role of `~`.

I will try to fix it, at least on the `pyplot_api` page, asap.

**Comment by Naereen** [comment]
I tried my best at locating and fixing the issues pointed out above.

**Comment by Naereen** [comment]
I closed #8080 which was not mergeable anymore, but this issue stays active and important.

**Comment by timhoffm** [comment]
As of 3.2.2 these docs render ok.
